PureLayout/PureLayout/ALView+PureLayout.m

```diff
@@ -184,24 +197,88 @@ - (NSLayoutConstraint *)autoPinEdgeToSuperviewSafeArea:(ALEdge)edge withInset:(C
         trailingAnchor = superview.trailingAnchor;
     }
 
+    if (edge == ALEdgeBottom || edge == ALEdgeRight || edge == ALEdgeTrailing) {
+        // The bottom, right, and trailing insets (and relations, if an inequality) are inverted to become offsets
+        inset = -inset;
+    }
     switch (edge) {
         case ALEdgeLeft:
-            constraint = [[self leftAnchor] constraintEqualToAnchor:leftAnchor constant:inset];
+            switch (relation) {
+                case NSLayoutRelationEqual:
+                    constraint = [[self leftAnchor] constraintEqualToAnchor:leftAnchor constant:inset];
+                    break;
+                case NSLayoutRelationLessThanOrEqual:
+                    constraint = [[self leftAnchor] constraintLessThanOrEqualToAnchor:leftAnchor constant:inset];
+                    break;
+                case NSLayoutRelationGreaterThanOrEqual:
+                    constraint = [[self leftAnchor] constraintGreaterThanOrEqualToAnchor:leftAnchor constant:inset];
+                    break;
+            }
             break;
         case ALEdgeRight:
-            constraint = [[self rightAnchor] constraintEqualToAnchor:rightAnchor constant:inset];
+            switch (relation) {
+                case NSLayoutRelationEqual:
+                    constraint = [[self rightAnchor] constraintEqualToAnchor:rightAnchor constant:inset];
+                    break;
+                case NSLayoutRelationLessThanOrEqual:
+                    constraint = [[self leftAnchor] constraintGreaterThanOrEqualToAnchor:rightAnchor constant:inset];
```

Comment:
@toohotz Is this mistake of `rightAnchor` ?

I think it expect:
```
constraint = [[self rightAnchor] constraintGreaterThanOrEqualToAnchor:rightAnchor constant:inset];
```

If not, Please tell me your intension.

Reply:
Good catch, copy pasted the initial one and forgot to change the `rightAnchor` to leftAnchor`, will fix it, thank you for catching.

Reply:
Thanks. Other code feel good, so I will merge as soon as it's fixed.

Reply:
No problem, done.